### PR TITLE
Knative Eventing/Sources CI changes

### PR DESCRIPTION
**Pull request by matzew**
* setup CI for the 0.4 release branch
* remove the master branch from CI (incl. periodic build)

Note: the master is not expected to be on CI for a while - just the release branches

### CODE REVIEW

File: ci-operator/jobs/openshift/knative-eventing-sources/openshift-knative-eventing-sources-release-0.4-presubmits.yaml [hunk 1]
```diff
@@ -28,8 +28,8 @@ presubmits:
         - name: CONFIG_SPEC
           valueFrom:
             configMapKeyRef:
-              key: openshift-knative-eventing-sources-master.yaml
-              name: ci-operator-master-configs
+              key: openshift-knative-eventing-sources-release-0.4.yaml
+              name: ci-operator-misc-conffigs
```
Comment (mgencur) [inline ci-operator/jobs/openshift/knative-eventing-sources/openshift-knative-eventing-sources-release-0.4-presubmits.yaml]: "conffigs" ? where does it come from? the CI is complaining about it. Maybe pulling the latest image and re-generating the configs would help?

File: ci-operator/jobs/openshift/knative-eventing-sources/openshift-knative-eventing-sources-release-0.4-presubmits.yaml [hunk 2]
```diff
@@ -81,8 +81,8 @@ presubmits:
...
```
Comment (mgencur) [inline ci-operator/jobs/openshift/knative-eventing-sources/openshift-knative-eventing-sources-release-0.4-presubmits.yaml]: "conffigs" again

File: ci-operator/jobs/openshift/knative-eventing/openshift-knative-eventing-release-0.4-postsubmits.yaml [hunk 3]
```diff
@@ -23,7 +23,7 @@ postsubmits:
         - name: CONFIG_SPEC
           valueFrom:
             configMapKeyRef:
-              key: openshift-knative-eventing-master.yaml
+              key: openshift-knative-eventing-release-0.4.yaml
               name: ci-operator-master-configs
```
Comment (mgencur) [inline ci-operator/jobs/openshift/knative-eventing/openshift-knative-eventing-release-0.4-postsubmits.yaml]: should this be "ci-operator-misc-configs" now that this is changed to release-0.4 branch?

Comment (matzew) [reply ci-operator/jobs/openshift/knative-eventing/openshift-knative-eventing-release-0.4-postsubmits.yaml]: @mgencur I did borrow from here:  https://github.com/openshift/release/blob/b5f8f0fb2399ec4bb2000a874d9b01c25db26106/ci-operator/jobs/openshift/knative-serving/openshift-knative-serving-release-0.4-postsubmits.yaml#L26

Comment (matzew) [reply ci-operator/jobs/openshift/knative-eventing/openshift-knative-eventing-release-0.4-postsubmits.yaml]: oh, I changed that now :-) @mgencur 

will add the script, ASAP ;-)

**Comment by petr-muller** [comment]
Bad bot did not update the CMs!

**Pull request merged**